### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by rupnikj**

### CODE REVIEW

File: src/distributed/topology_local_wrapper.ts [hunk 1]
```diff
@@ -8,43 +8,69 @@ import * as serialize_error from "serialize-error"
...
+            self.clearPingInterval();
             self.killProcess(intf.ChildExitCode.unhandeled_error, e);
         });
-        process.on('SIGINT', () => {
-            log.logger().warn(self.log_prefix + "Received SIGINT, this process id = " + process.pid);
-            self.shutdown();
+        this.process.on('SIGINT', () => {
+            log.logger().warn(self.log_prefix + "Received SIGINT, this process id = " + self.process.pid);
+            if (!self.topology_local) {
+                this.exitNonInit("Shutdown", intf.ChildMsgCode.response_shutdown,
```
Comment (bergloman) [inline src/distributed/topology_local_wrapper.ts]: using `this` - it works, but is dangerous, IMO

File: src/distributed/topology_local_wrapper.ts [hunk 2]
```diff
@@ -8,43 +8,69 @@ import * as serialize_error from "serialize-error"
...
+                self.shutdown();
+            }
         });
-        process.on('SIGTERM', () => {
-            log.logger().warn(self.log_prefix + "Received SIGTERM, this process id = " + process.pid);
-            self.shutdown();
+        this.process.on('SIGTERM', () => {
+            log.logger().warn(self.log_prefix + "Received SIGTERM, this process id = " + self.process.pid);
+            if (!self.topology_local) {
+                this.exitNonInit("Shutdown", intf.ChildMsgCode.response_shutdown,
```
Comment (bergloman) [inline src/distributed/topology_local_wrapper.ts]: using `this` - it works, but is dangerous, IMO

File: src/distributed/topology_local_wrapper.ts [hunk 3]
```diff
@@ -126,30 +177,41 @@ class TopologyLocalWrapper {
...
                 }
             });
         }
         if (msg.cmd === intf.ParentMsgCode.shutdown) {
             if (!self.topology_local) {
-                let s = `Shutdown called in the child process, but the topology hasn't been initialized yet.`;
-                log.logger().error(self.log_prefix + s);
-                self.sendToParent(intf.ChildMsgCode.response_shutdown, { err: new Error(s) });
-                self.killProcess(intf.ChildExitCode.shutdown_notinit_error, new Error(s));
+                this.exitNonInit("Shutdown", intf.ChildMsgCode.response_shutdown,
```
Comment (bergloman) [inline src/distributed/topology_local_wrapper.ts]: using `this` - it works, but is dangerous, IMO

File: src/topology_local.ts [hunk 4]
```diff
@@ -178,38 +181,44 @@ export class TopologyLocal {
...
         for (let spout of this.spouts) {
             spout.pause();
         }
-        this.isRunning = false;
         return callback();
     }
 
     /** Sends shutdown signal to all child processes */
     shutdown(callback: intf.SimpleCallback) {
+        //callback = this.tryCallback(callback);
```
Comment (bergloman) [inline src/topology_local.ts]: not used? then you can delete it.

**Comment by rupnikj** [comment]
Updated according to comments

**Pull request merged**